### ISSUE TITLE
create-react-app production build fails

### ISSUE THREAD

**Issue by peteyy** [solkimicreb/react-easy-state#3]
Command: `npm run build`

Output:

```
> react-scripts build
3:44:35 PM: Creating an optimized production build...
3:44:43 PM: Failed to compile.
3:44:43 PM: Failed to minify the code from this file: 

./node_modules/@nx-js/observer-util/src/observer.js:8 

Read more here: http://bit.ly/2tRViJ9
3:44:43 PM:
3:44:43 PM: Build complete: exit code: 1
3:44:43 PM:
3:44:44 PM: Cleaning up docker container
3:44:44 PM: npm
3:44:44 PM:
3:44:44 PM: ERR!
3:44:44 PM: Linux 3.19.0-66-generic
3:44:44 PM: npm
3:44:44 PM:
3:44:44 PM: ERR!
3:44:44 PM: argv "/opt/buildhome/.nvm/versions/node/v6.11.2/bin/node" "/opt/buildhome/.nvm/versions/node/v6.11.2/bin/npm" "run" "build"
3:44:44 PM: npm
3:44:44 PM: ERR! node
3:44:44 PM: v6.11.2
npm
3:44:44 PM:
3:44:44 PM: Error running command: Build script returned non-zero exit code: 1
3:44:45 PM: ERR! npm v3.10.10
3:44:45 PM: npm
3:44:45 PM: ERR!
3:44:45 PM: An error occurred while building the site, skipping the deploy and cache refresh.
3:44:45 PM: code ELIFECYCLE
3:44:45 PM: npm ERR!
3:44:45 PM: rep-counter-client@0.1.0 build: `react-scripts build`
npm
3:44:45 PM: ERR!
3:44:45 PM: Build script returned non-zero exit code: 1
3:44:45 PM: Exit status 1
npm
3:44:45 PM: ERR!
3:44:45 PM: npm
3:44:45 PM: ERR!
3:44:45 PM: Failed at the rep-counter-client@0.1.0 build script 'react-scripts build'.
3:44:45 PM: npm ERR!
3:44:45 PM: Make sure you have the latest version of node.js and npm installed.
3:44:46 PM: npm ERR!
3:44:46 PM: If you do, this is most likely a problem with the rep-counter-client package,
npm
3:44:46 PM:
3:44:46 PM: ERR!
3:44:46 PM: not with npm itself.
npm
3:44:46 PM:
3:44:46 PM: ERR!
3:44:46 PM: Tell the author that this fails on your system:
npm ERR! react-scripts build
npm ERR!
3:44:46 PM: You can get information on how to open an issue for this project with:
3:44:46 PM: npm ERR!
3:44:46 PM: Finished processing build request in 49.305966188s
3:44:46 PM: npm bugs rep-counter-client
3:44:46 PM: npm
3:44:46 PM: ERR!
3:44:47 PM: Or if that isn't available, you can get their info via:
npm
3:44:47 PM: ERR!
3:44:47 PM: npm owner ls rep-counter-client
3:44:47 PM: npm
3:44:47 PM: ERR!
3:44:47 PM: There is likely additional logging output above.
3:44:47 PM:
3:44:47 PM: npm
3:44:47 PM: ERR! Please include the following file with any support request:
npm
3:44:47 PM: ERR! /opt/build/repo/npm-debug.log
3:44:47 PM: Cached NPM modules
```

**Comment by solkimicreb** [comment]
Thanks for the issue!

Both the commonJS and the esm builds are transpiled down only too the minimum supported platforms, which are all ES6. Webpack uses uglifyJS as its default minifier, which does not support ES6 yet. This means that the minification will fail even though the platform support is there.

To work around this I can think of 2 things:

- Ask users to use an es6 aware minifier. This is a no-go as I would like to remain compatible with React Create App and default webpack config.

- Distribute React Easy State in a pseudo-es5 format, even though it won't make platform support wider. It would solve the issue, but it would also introduce unnecessary transpilation. I would go with this until UglifyJS starts to support es6.

**Comment by solkimicreb** [comment]
Fixed in v2.0.3. Thanks for the issue 🙂
